src/components/form/form.component.js

```diff
@@ -0,0 +1,511 @@
+  cancel: PropTypes.bool,
+
+  /** Custom function that is called immediately after the form validates */
+  afterFormValidation: PropTypes.func,
+
+  /** Custom function that is called immediately before the form validates */
+  beforeFormValidation: PropTypes.func,
+
+  /** Alignment of submit button */
+  buttonAlign: PropTypes.string,
+
+  /** Determines if the form is in a saving state */
+  saving: PropTypes.bool,
+
+  /** Enables the sticky footer. */
+  stickyFooter: PropTypes.bool,
+
+  /** Applies additional padding to the sticky footer, useful to align buttons. */
+  stickyFooterPadding: PropTypes.string,
+
+  /** If true, will validate the form on mount */
+  validateOnMount: PropTypes.bool,
+
+  /** If true, will disable the savebutton when clicked */
+  autoDisable: PropTypes.bool,
+
+  /** Text for the cancel button */
+  cancelText: PropTypes.string,
+
+  /** Properties for the cancel button */
+  cancelButtonProps: PropTypes.object,
+
+  /** Text for the save button */
+  saveText: PropTypes.string,
+
+  /** Properties for the save button */
+  saveButtonProps: PropTypes.object,
+
+  /** Custom function for Cancel button onClick */
+  onCancel: PropTypes.func,
+
+  /** Hide or show the save button */
+  save: PropTypes.bool,
+
+  /** Additional actions rendered next to the save and cancel buttons */
+  additionalActions: PropTypes.node, // eslint-disable-line react/no-unused-prop-types
+
+  /** Additional actions rendered and aligned left to the save and cancel buttons */
+  leftAlignedActions: PropTypes.node, // eslint-disable-line react/no-unused-prop-types
+
+  /** Additional actions rendered and aligned right to the save and cancel buttons */
+  rightAlignedActions: PropTypes.node, // eslint-disable-line react/no-unused-prop-types
+
+  /** Custom callback for when form will submit */
+  onSubmit: PropTypes.func,
+
+  /** Override Save Button */
+  customSaveButton: PropTypes.node,
+
+  /** Child elements */
+  children: PropTypes.node,
+
+  /** Hide or show the summary */
+  showSummary: PropTypes.bool,
+
+  /** A function used to handle form validation */
+  validate: PropTypes.func,
+
+  /** Determines if the form is a validating state and should be disabled from submiting */
+  isValidating: PropTypes.bool,
+
+  /** The total number of errors present in the form */
+  errorCount: PropTypes.number,
+
+  /** The total number of warnings present in the form */
+  warningCount: PropTypes.number,
+
+  /** The total number of infos present in the form */
+  infoCount: PropTypes.number,
+
+  /** Passed down when Form parent is dialog */
+  fixedBottom: PropTypes.bool,
+
+  /** Sets children's label alignment */
+  isLabelRightAligned: PropTypes.bool
+};
+
+FormWithoutValidations.defaultProps = {
+  buttonAlign: 'right',
+  cancel: true,
+  unsavedWarning: true,
+  save: true,
+  saving: false,
+  validateOnMount: false,
+  customSaveButton: null,
+  showSummary: true
+};
+
+FormWithoutValidations.childContextTypes = {
```

Comment:
`childContextTypes` and `contextTypes` are part of [React's legacy context API](https://reactjs.org/docs/legacy-context.html). Is it OK to be using them here?

Reply:
:white_check_mark: